[rdy] Add homes for rat families in dirty hospitals

Alberth289346 opened the pull request:
Let there be holes in the walls

TheCycoONE commented:
Rathole is one word. This patch mixes ```RatHole``` and ```Rathole``` (and uses ```rat_hole```)

TheCycoONE commented on CorsixTH/Src/th_lua_map.cpp:
```diff
@@ -832,6 +832,40 @@ static int l_map_is_parcel_purchasable(lua_State *L)
+                continue;
+
+            tile_count++;
+            for(auto iter = pNode->objects.begin(); iter != pNode->objects.end(); iter++)
```
Always use braces

TheCycoONE commented on CorsixTH/Src/th_lua_map.cpp:
```diff
@@ -832,6 +832,40 @@ static int l_map_is_parcel_purchasable(lua_State *L)
+    double tile_count = 0;
+    double litter_count = 0;
+    for (int x = 0; x < pMap->getWidth(); x++)
+        for (int y = 0; y < pMap->getHeight(); y++)
```
Use braces

Alberth289346 commented:
All fixed

Alberth289346 commented:
rebased, updated game version number

Alberth289346 commented:
drop removing holes by cleaning the hospital, and add removal by placing object on top of it.

Alberth289346 commented:
New version, ratholes get removed when building objects, and rooms, generalizing the world: remove litter functions. Also, you can use objects and rooms to remove ratholes

TheCycoONE commented on CorsixTH/Lua/hospital.lua:
```diff
@@ -1987,6 +2037,121 @@ function Hospital:checkDiseaseRequirements(disease)
+    if rathole.x == x and rathole.y == y and rathole.wall == wall then return end
+  end
+
+  local hole = {x=x, y=y, wall=wall, parcel=parcel}
```
Spaces around assignment operators (consistency)

TheCycoONE commented on CorsixTH/Lua/hospital.lua:
```diff
@@ -1987,6 +2037,121 @@ function Hospital:checkDiseaseRequirements(disease)
+    if rathole.x == hole.x and rathole.y == hole.y and rathole.wall == hole.wall then
+      table.remove(self.ratholes, i)
+      if rathole.object then self.world:destroyEntity(rathole.object) end
+      break
```
Maybe an extra newline, or move the body of the if onto the next line. Just trying to separate the break so it doesn't look like it might belong with the previous if.

TheCycoONE commented:
Looks like a fine approach.

Alberth289346 commented:
Fixed spaces
```` lua
local hole = {x = x, y = y, wall = wall, parcel = parcel}
````

indenting the if didn't look right, so I added an empty line too
```` lua
    if rathole.x == hole.x and rathole.y == hole.y and rathole.wall == hole.wall then
      table.remove(self.ratholes, i)
      if rathole.object then
        self.world:destroyEntity(rathole.object)
      end

      break
    end
````

That should be enough :)
